mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,506 @@
+        rescale(data_am, times, baseline, baseline_kind, copy=False)
+        msk_time = _time_mask(times, *use_times)
+        data_am, data_ph = [i[..., msk_time] for i in [data_am, data_ph]]
+
+        # Stack epochs to a single trace if specified
+        if concat_epochs is True:
+            ev_unique = np.unique(ev_grouping)
+            concat_data = []
+            for i_ev in ev_unique:
+                msk_events = ev_grouping == i_ev
+                concat_data.append([np.hstack(i[msk_events])
+                                    for i in [data_am, data_ph]])
+            data_am, data_ph = zip(*concat_data)
+    else:
+        data_ph, data_am = [data_ph], [data_am]
+
+    n_ep = len(data_ph)
+    pac = np.zeros([n_ep, len(ixs_new)])
+    pbar = ProgressBar(n_ep)
+    for iep, (ep_ph, ep_am) in enumerate(zip(data_ph, data_am)):
+        for iix, (i_ix_ph, i_ix_am) in enumerate(ixs_new):
+            # f_phase and f_amp won't be used in this case
+            pac[iep, iix] = func(ep_ph[i_ix_ph], ep_am[i_ix_am],
+                                 f_phase, f_amp, filterfn=False)
+        pbar.update_with_increment_value(1)
+    if return_data:
+        return pac, data_ph, data_am
+    else:
+        return pac
+
+
+def _pre_filter_ph_am(data, sfreq, ixs, f_ph, f_am, filterfn=None,
+                      npad=None, hi_phase=False, scale_amp_func=None,
+                      kws_filt=None):
+    """Filter for phase/amp only once for each channel."""
+    from ..externals.pacpy.pac import _range_sanity
+    from scipy.signal import hilbert
+    filterfn = band_pass_filter if filterfn is None else filterfn
+    kws_filt = dict() if kws_filt is None else kws_filt
+    _range_sanity(f_ph, f_am)
+    ix_ph = np.atleast_1d(np.unique(ixs[:, 0]))
+    ix_am = np.atleast_1d(np.unique(ixs[:, 1]))
+
+    # For padding if input is shape of raw
+    n_times = data.shape[-1]
+    if npad == 'auto':
+        next_pow_2 = int(np.ceil(np.log2(n_times)))
+        npad = 2**next_pow_2 - n_times
+    elif not isinstance(npad, int):
+        raise ValueError('npad must be "auto" or a positive integer')
+
+    data_ph = np.hstack([data[ix_ph, :], np.zeros([len(ix_ph), npad])])
+    data_ph = filterfn(data_ph, sfreq, *f_ph, **kws_filt)
+    data_ph = np.angle(hilbert(data_ph))[..., :n_times]
+    ix_map_ph = {ix: i for i, ix in enumerate(ix_ph)}
+
+    data_am = np.hstack([data[ix_am], np.zeros([len(ix_am), npad])])
+    data_am = filterfn(data_am, sfreq, *f_am, **kws_filt)
+    data_am = np.abs(hilbert(data_am))
+    if hi_phase is True:
+        data_am = filterfn(data_am, sfreq, *f_ph, **kws_filt)
+        data_am = np.angle(hilbert(data_am))
+    data_am = data_am[..., :n_times]
+    ix_map_am = {ix: i for i, ix in enumerate(ix_am)}
+
+    if scale_amp_func is not None:
+        data_am = scale_amp_func(data_am, axis=-1)
+    return data_ph, data_am, ix_map_ph, ix_map_am
+
+
+def _filter_ph_am(xph, xam, f_ph, f_am, sfreq, filterfn=None, kws_filt=None):
```

Comment:
ah thanks for catching it - it got deprecated with the "pre_filter" function that operates on lots of channels